src/v/raft/consensus.cc

```diff
@@ -1774,7 +1774,17 @@ ss::future<> consensus::do_hydrate_snapshot(storage::snapshot_reader& reader) {
         update_follower_stats(metadata.latest_configuration);
         return _configuration_manager
           .add(_last_snapshot_index, std::move(metadata.latest_configuration))
-          .then([this, delta = metadata.log_start_delta] {
+          .then([this, delta = metadata.log_start_delta]() mutable {
+              if (delta < offset_translator_delta(0)) {
+                  delta = offset_translator_delta(
+                    _configuration_manager.offset_delta(_last_snapshot_index));
```

Comment:
I wonder if we should use delta from the offset translator itself here? Or leave a comment detailing why we expect only configuration batches to contribute to offset translation if the delta in the snapshot is empty.

Reply:
you are right, i did not realize that there are other batch types, but otoh did we have different batch types in previous versions ?

Reply:
In 21.9 - probably no. In 21.10 - yes

Reply:
exactly v21.10, have delta already,